index.js

```diff
@@ -43,25 +41,61 @@ const getHighWaterMark = (streams, objectMode) => {
 	return Math.max(...highWaterMarks);
 };
 
-// Number of times `passThroughStream.on()` is called:
-//  - once per stream due to `stream.pipe(passThroughStream)`
+class MergedStreams extends PassThroughStream {
+	#streams = [];
+	#ended = new Set([]);
+
+	add(stream) {
+		if (!this.writable) {
+			throw new TypeError('The merged stream has already ended.');
+		}
+
+		validateStream(stream);
+		this.#streams.push(stream);
+		this.setMaxListeners(this.getMaxListeners() + PASSTHROUGH_LISTENERS_COUNT);
+		endWhenStreamsDone(this, stream, this.#streams, this.#ended);
+		stream.pipe(this, {end: false});
+	}
+
+	remove(stream) {
+		const streamIndex = this.#streams.indexOf(stream);
```

Comment:
`Array#includes`?

Reply:
Right! 
I was initially using `streamIndex` for something else, but not anymore, so `.includes()` should be used now.
Done. :+1: